### PR TITLE
(v0.6.0) Simplify boundary condition construction

### PR DESCRIPTION
**Pull request by navidcy**
*Description edited by Ashley to reflect discussion below*
This PR aims to simplify the boundary condition construction in the simple case of four boundaries that are parallel to lines of constant longitude and latitude, as per the discussion in https://github.com/COSIMA/regional-mom6/issues/174.

To do this, we add a wrapper function that just calls the `simple_boundary` method for each cardinal direction requested. 

This PR includes a breaking API change since `rectangular_boundary` is renamed to `rectangular_boundaries` and also some of its args changed.

Closes https://github.com/COSIMA/regional-mom6/issues/174

### CODE REVIEW

File: regional_mom6/regional_mom6.py [hunk 1]
```diff
@@ -874,11 +879,23 @@ def initial_condition(
...
-        print("done setting up initial condition.")
 
         self.ic_eta = eta_out
         self.ic_tracers = tracers_out
         self.ic_vels = vel_out
+
+        # Now iterate through our four boundaries
+        for i, orientation in enumerate(boundaries, start=1):
+            self.rectangular_boundary(
+                glorys_path / (orientation + "_unprocessed.nc"),
```
Comment (navidcy) [inline regional_mom6/regional_mom6.py:890]: what this should be? definitely not `glorys_path` right?

**Comment by ashjbarnes** [comment]
In this PR you're changing the boundary condition function not the initial condition right? I still think that the boundary and initial conditions should be kept as separate functions. If not, then we'll should rename it so that the "initial condition" function doesn't do the boundary right?

File: regional_mom6/regional_mom6.py (hunk 1 above)
Comment (ashjbarnes) [reply regional_mom6/regional_mom6.py:890]: You could assume that the user has put the boundary forcing files in the same folder as the initial condition I guess? This is the path that's passed to the 'initial condition' function

**Comment by navidcy** [comment]
Oh my bad

**Comment by ashjbarnes** [comment]
This PR will require a new tag release as it's not backwards compatible. Also need to update [the access om2 example](https://github.com/COSIMA/cosima-recipes/pull/338) with the changes.

**Comment by ashjbarnes** [comment]
I'm going to approve the PR because it was Navid's originally. Wait on merging until @navidcy is happy and we also fix up the access om2 example to match simultaneously

**Review by ashjbarnes — APPROVED** [review]

**Comment by navidcy** [comment]
I'd like to merge this and then we tag a new release -- how does this sound @ashjbarnes?

**Comment by ashjbarnes** [comment]
Great! Need to reflect changes in [access om2 example](https://github.com/COSIMA/cosima-recipes/pull/338)

**Pull request merged**